File: modules/modDice.py
# modDice.py
import re
from random import randint

from modules.mod import mod

class module(mod):
	def __init__(self):
		super().__init__()
		self.description = "Roll a Dice of any size!"
		self.how = "`?dice {value}` (value mayb start with the letter 'd' e.g. d20)"

	async def run(self, *args):

		# No arguments given; empty args: ([],)
		if len(args[0]) != 0:
			return self.how

		dMax = re.search(r"^(|d|D)\d+", str(args[0][0]))
		if not dMax:
			return self.how

		roll: int = -1
		if dMax.group(0)[0] == 'd':
			roll = int(dMax.group(0)[1:])
		else:
			roll = int(dMax.group(0))

		if roll <= 1:
			return self.how
		else:
			return f"Rolled: `{randint(1, roll)}`"

File: bot/modules/modDice.go
package modules

import (
	"fmt"
	"math/rand"
	"strconv"
)

var ModDice Module = Module{
	Keys: []string{
		"d",
		"dice",
	},

	Description: "Roll a Dice of any size!",
	How:         "`?d {size}` or `?dice {size}` | `{size}` must be a number greater than 1 | Ranges from `1` to `9223372036854775807 (2^63 - 1)`",

	Run: func(args []string) (resp string) {
		resp = ""

		if len(args) == 0 {
			resp = "Not enough arguments given! Expected: 1; Got: 0"
		} else if args[0] == "" {
			resp = "Not enough arguments given! Expected: 1; Got: 0"
		}

		if resp == "" {
			if roll, err := strconv.ParseInt(args[0], 10, 64); err != nil {
				resp = "Could not parse given argument"
			} else {
				if roll < 1 {
					resp = "Argument `{size}` must be a number geater than 1"
				} else {
					resp = fmt.Sprintf("Rolled `%v`!", rand.Intn(int(roll))+1)
				}
			}
		}

		return
	},
}
